### PR TITLE
Set gt_cache directory at module import time and remove future stencil

### PR DESCRIPTION
**Pull request by jdahm**
## Purpose

To avoid test failures on 54 ranks due to distributed filesystem causing stencils not to be found after compiling.

## Code changes:

- Sets the gt_cache directory when loading `pace.dsl` to a unique directory for each MPI rank.

### CODE REVIEW

File: dsl/pace/dsl/__init__.py [hunk 1]
```diff
@@ -1 +1,10 @@
+import gt4py.config
+
+from pace.util.mpi import MPI
+
 from .stencil import FrozenStencil, StencilConfig, StencilFactory
+
+
+if MPI is not None:
+    gt4py.config.cache_settings["root_path"] = "."
+    gt4py.config.cache_settings["dir_name"] = f".gt_cache_{MPI.COMM_WORLD.Get_rank()}"
```
Comment (FlorianDeconinck) [inline dsl/pace/dsl/__init__.py:10]: Should we re-use gt4py env var?
```
    "dir_name": os.environ.get("GT_CACHE_DIR_NAME", ".gt_cache"),
    "root_path": os.environ.get("GT_CACHE_ROOT", os.path.abspath(".")),
 ```

**Review by FlorianDeconinck — APPROVED** [review]

**Comment by jdahm** [comment]
The only test that failed here timed out. This should be ready to go!

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by elynnwu** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by elynnwu** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Pull request merged**